src/controllers/flipTimerController.js

```diff
@@ -0,0 +1,96 @@
+class FlipTimer {
+  constructor({ selector, start, store }) {
+      this.selector = selector
+      this.start = start
+      this.store = store
+
+      this.init(selector, start)
+  }
+
+  init(selector, start) {
+      const container = $(selector)
+
+      const endDate = this.getEndDate(container)
+
+      if (!endDate) {
+          return;
+      }
+
+      const expiresTime = this.calculateExpiresTime(endDate)
+
+      container.innerHTML = TIMER_CONFIG.map(function ({ type, accessor }) {
+          const currentExpires = expiresTime[accessor]
+
+          return `<li class="flip-item-${type}">${currentExpires}</li>`
+      }).join('')
+
+      renderTimer(jQuery, start, store)
+  }
+
+  getEndDate(container) {
+      if (!container) {
+          return;
+      }
+
+      const endDate = container.dataset.endDate
+      const formattedDate = new Date(endDate)
+
+      if (!endDate || !formattedDate) {
+          return new Date()
+      }
+
+      return formattedDate
+  }
+
+  calculateExpiresTime(endDate) {
+      const ONE_SECOND = 1000
+      const ONE_MINUTE = ONE_SECOND * 60
+      const ONE_HOUR = ONE_MINUTE * 60
+      const ONE_DAY = 24 * 60 * 60 * 1000
+
+      const todayDate = new Date()
+      const dateDifference = Math.abs(todayDate - endDate)
+
+      const daysLeft = Math.max(0, Math.floor(dateDifference / ONE_DAY))
+      const hoursLeft = Math.max(0, Math.floor((dateDifference / ONE_HOUR) % 24))
+      const minutesLeft = Math.max(
+          0,
+          Math.floor((dateDifference / ONE_MINUTE) % 60)
+      )
+      const secondsLeft = Math.max(
+          0,
+          Math.floor((dateDifference / ONE_SECOND) % 60)
+      )
+
+      return {
+          daysLeft,
+          hoursLeft,
+          minutesLeft,
+          secondsLeft,
+      }
+  }
+
+  showIsHidden() {
+      setTimeout(() => {
+          this.showIsHidden()
+      }, 200)
+  }
+}
+
+FlipTimer.unmountTimers = function () {
+  document.querySelectorAll('.flip-clock-container')
+      .forEach((container) => {
+          container.innerHTML = ''
+      })
+}
+
+FlipTimer.addTimer = function (selector, start) {
+  return new FlipTimer({ selector, start, store })
```

Comment:
Can we upgrade this solution?

let's move these timers to the local **const** variable, **state** or **this**.
After, let's use stored timers in the **unmountTimers** function and exclude querySelectorAll usage to avoid side effects.